packages/fuels-code-gen/src/program_bindings/abigen/bindings/contract.rs

```diff
@@ -62,8 +64,25 @@ pub(crate) fn contract_bindings(
                 self.account.clone()
             }
 
-            pub fn with_account<U: ::fuels::accounts::Account>(&self, account: U) -> ::fuels::types::errors::Result<#name<U>> {
-                ::core::result::Result::Ok(#name { contract_id: self.contract_id.clone(), account, log_decoder: self.log_decoder.clone()})
+            pub fn with_account<U: ::fuels::accounts::Account>(&self, account: U)
+            -> ::fuels::types::errors::Result<#name<U>> {
+                ::core::result::Result::Ok(
+                #name {
+                        contract_id: self.contract_id.clone(),
+                        account,
+                        log_decoder: self.log_decoder.clone(),
+                        encoder_config: self.encoder_config.clone()
+                })
+            }
+
+            pub fn with_encoder_config(&self, encoder_config: ::fuels::core::codec::EncoderConfig)
+            -> #name::<T> {
+                #name {
+                        contract_id: self.contract_id.clone(),
+                        account: self.account(),
+                        log_decoder: self.log_decoder.clone(),
+                        encoder_config,
+                }
```

Comment:
I would use the builder pattern here and take ownership. If the user wants to keep a clone he should explicitly do so. Btw this is what we do for `script_instances`

```suggestion
            pub fn with_encoder_config(mut self, encoder_config: ::fuels::core::codec::EncoderConfig)
            -> #name::<T> {
                self.encoder_config = encoder_config;

                self
            }
```